### PR TITLE
Add `external_metrics_provider.chunk_size` parameter

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add `external_metrics_provider.chunk_size` parameter to decrease the number of queries batched by the agent and sent to Datadog.

### Motivation

Attempt to mitigate metrics server side timeout errors like:

```
2021-11-24 10:52:16 UTC | CLUSTER | ERROR | (pkg/util/kubernetes/autoscalers/datadogexternal.go:79 in queryDatadogExternal) | Error while executing metric query ... truncated... API returned error: Query timed out
```

Maybe sending less queries in a batch will help not timing out.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/configurable_chunk_size-b22644b3b2f3311b.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The cluster agent has an external metrics provider feature to allow using datadog queries in Kubernetes HorizontalPodAutoscalers.
```
Comment (kayayarai) [inline releasenotes/notes/configurable_chunk_size-b22644b3b2f3311b.yaml:11]:
```suggestion
    The cluster agent has an external metrics provider feature to allow using Aatadog queries in Kubernetes HorizontalPodAutoscalers.
```

File: releasenotes/notes/configurable_chunk_size-b22644b3b2f3311b.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    It sometimes faces issues like:
+
+      2022-01-01 01:01:01 UTC | CLUSTER | ERROR | (pkg/util/kubernetes/autoscalers/datadogexternal.go:79 in queryDatadogExternal) | Error while executing metric query ... truncated... API returned error: Query timed out
+
+    In order to mitigate this problem the ``external_metrics_provider.chunk_size`` parameter has been introduced to reduce the number of queries that are batched by the agent and sent together to Datadog.
```
Comment (kayayarai) [inline releasenotes/notes/configurable_chunk_size-b22644b3b2f3311b.yaml:16]:
```suggestion
    To mitigate this problem, use the new ``external_metrics_provider.chunk_size`` parameter to reduce the number of queries that are batched by the Agent and sent together to Datadog.
```

**Review by kayayarai — APPROVED** [review]
minor edits

**Review by sgnn7 — APPROVED** [review]

File: pkg/config/config.go [hunk 3]
```diff
@@ -1218,6 +1222,7 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 	useHostEtc(config)
 	loadProxyFromEnv(config)
 	SanitizeAPIKeyConfig(config, "api_key")
+	sanitizeExternalMetricsProviderChunkSize(config)
```
Comment (vboulineau) [inline pkg/config/config.go:1225]: We have the `overrideFuncs` to register functions like that

Comment (L3n41c) [reply pkg/config/config.go]: Something like this: https://github.com/DataDog/datadog-agent/pull/10781/commits/5a2915f5acd4eae9eeb794d8448a96dcfd05fa95 ?

**Review by vboulineau — APPROVED** [review]

**Pull request merged**